### PR TITLE
windowing/gbm: avoid crashing if ttyname returns NULL

### PR DESCRIPTION
**Pull request by lrusak**
This will avoid us hard crashing when `ttyname` return `NULL`. This happens when STDIN isn't conntected to any tty and is defaulted to `/dev/null`

see, http://man7.org/linux/man-pages/man3/ttyname.3.html
and, https://www.freedesktop.org/software/systemd/man/systemd.exec.html#Logging%20and%20Standard%20Input/Output

### CODE REVIEW

**Comment by Rechi** [comment]
jenkins build this please

File: xbmc/windowing/gbm/VTUtils.cpp [hunk 1]
```diff
@@ -23,7 +23,21 @@ using namespace KODI::WINDOWING::GBM;
 
 bool CVTUtils::OpenTTY()
 {
-  m_ttyDevice = ttyname(STDIN_FILENO);
+  const char* ttyDevice = ttyname(STDIN_FILENO);
```
Comment (a1rwulf) [inline xbmc/windowing/gbm/VTUtils.cpp]: Can we change this to `ttyname_r` while we are at it?

Comment (lrusak) [reply xbmc/windowing/gbm/VTUtils.cpp]: Does it really add any benefit?

Comment (a1rwulf) [reply xbmc/windowing/gbm/VTUtils.cpp]: It's the reentrant version.
I usually prefer them in utils code, just to skip thinking about threading issues.

Comment (lrusak) [reply xbmc/windowing/gbm/VTUtils.cpp]: Updated

**Comment by lrusak** [comment]
closing due to #14776

**Pull request closed without merging**